src/minoltamn_int.cpp

```diff
@@ -1657,12 +1657,9 @@ std::ostream& printMinoltaSonyLensID(std::ostream& os, const Value& value, const
 
   // #1145 - respect lenses with shared LensID
   uint32_t index = value.toUint32();
-  const LensIdFct* lif = find(lensIdFct, index);
-  if (lif && metadata) {
-    if (lif->fct_)
-      return lif->fct_(os, value, metadata);
-  }
-
+  for (auto&& fct : lensIdFct)
+    if (metadata && fct == index && fct.fct_)
```

Comment:
Either this is wrong or the original code is wrong. I would think it's supposed to be

```
for (auto&& [idx, fct] : lensIdFct)
    if (metadata && idx == index && fct)
        return fct(os, value, metadata);
```

Reply:
Yeah it is. Fixed.

Reply:
Oh I see. The struct has an operator==. Although probably pointless with structured bindings.